wemake_python_styleguide/visitors/ast/annotations.py

```diff
@@ -59,6 +60,23 @@ def _check_for_literal_none(self, node: ast.Subscript) -> None:
         if slice_index.value.value is None:
             self.add_violation(LiteralNoneViolation(node))
 
+    def _check_for_nested_annotations(self, node: ast.Subscript) -> None:
+        annotation_name = cast(ast.Name, node.value)
+        slice_index = cast(ast.Index, node.slice)
+
+        if annotation_name.id not in {'Literal', 'Union'}:
```

Comment:
Let's move it to the class level `_flat_types`. We also need to check `Annotated` type